### PR TITLE
TS common-adapters: Image through RadioButton

### PR DESCRIPTION
**Pull request by buoyad**
r? @keybase/react-hackers

### CODE REVIEW

File: shared/common-adapters/kb-text-input.native.tsx [hunk 1]
```diff
@@ -44,31 +47,48 @@ if (isAndroid) {
 
       const textContainer = (
         <NativeTextInput
+          // @ts-ignore no RN types
```
Comment (MarcoPolo) [inline shared/common-adapters/kb-text-input.native.tsx]: Are the `@types/react-native` any good?

File: shared/common-adapters/new-input.tsx [hunk 2]
```diff
@@ -88,19 +88,14 @@ class ReflessNewInput extends React.Component<DefaultProps & Props, State> {
...
-      keyboardType?: KeyboardType,
-      textType?: TextType,
-    } & Props,
-    {
-      forwardedRef: React.Ref<typeof PlainInput>,
-    }
-  >,
-  PlainInput
->((props, ref) => <ReflessNewInput {...props} forwardedRef={ref} />)
+type FRefProps = {
```
Comment (MarcoPolo) [inline shared/common-adapters/new-input.tsx]: Unused?

File: shared/common-adapters/plain-input.d.ts [hunk 3]
```diff
@@ -1,23 +1,21 @@
...
-import type {TextType} from './text'
+import {StylesCrossPlatform} from '../styles'
+// @ts-ignore not converted
+import {TextType} from './text'
 
 export type KeyboardType =
   | 'default'
   | 'email-address'
-  | 'numeric'
+  | 'numeric' // iOS only
```
Comment (MarcoPolo) [inline shared/common-adapters/plain-input.d.ts]: These comments are in the wrong place. (Sorry this is hard for the parser to pick up properly)

File: shared/common-adapters/plain-input.d.ts [hunk 4]
```diff
@@ -51,67 +49,70 @@ export type TextContentType =
...
+export type Selection = {
+  start: number
+  end: number
+}
 
-export type Props = {|
-  autoFocus?: boolean,
+export type Props = {
+  autoFocus?: boolean
   // Enable if you want this to always have focus (desktop only)
```
Comment (MarcoPolo) [inline shared/common-adapters/plain-input.d.ts]: Check these comments too

File: shared/common-adapters/popup-dialog-hoc.tsx [hunk 5]
```diff
@@ -1,14 +1,13 @@
...
 type HocExtractProps = {
-  onClosePopup: () => void,
+  onClosePopup: () => void
 }
 
-function popupDialogHoc<Config: {} & HocExtractProps>(
-  Component: React.AbstractComponent<$Diff<Config, HocExtractProps>>
-): React.AbstractComponent<Config & HocExtractProps> {
+function popupDialogHoc<Config extends HocExtractProps>(
+  Component: React.ComponentType<Pick<Config, Exclude<keyof Config, keyof HocExtractProps>>>
```
Comment (MarcoPolo) [inline shared/common-adapters/popup-dialog-hoc.tsx]: Prefer Omit?

`ComponentType<Omit<Config, keyof HocExtractProps>>`

File: shared/common-adapters/kb-text-input.native.tsx (hunk 1 above)
Comment (buoyad) [reply shared/common-adapters/kb-text-input.native.tsx]: Tried locally a bit and they seem good, we run into this issue: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/26218. I wonder if there's a way to work around without having to run `tsc` twice.

**Comment by buoyad** [comment]
Thanks @MarcoPolo!

**Pull request merged**